### ISSUE TITLE
yCursorPlugin exception when ySyncPlugin not initialized

### ISSUE THREAD

**Issue by khrvati** [yjs/y-prosemirror#156]
### Issue

I'm using **ySyncPlugin** and **yCursorPlugin** together, to keep my editor up-to-date and to share cursor information through awareness. Sometimes, when a client is already connected to the same room and sharing their cursor position, trying to connect with a second client causes an exception

```
Uncaught TypeError TypeError: Cannot read properties of undefined (reading 'nodeSize')
    at relativePositionToAbsolutePosition (/Users/krunohrvatinic/ProjectRootDirName/node_modules/y-prosemirror/src/lib.js:183:63)
    at <anonymous> (/Users/krunohrvatinic/ProjectRootDirName/node_modules/y-prosemirror/src/plugins/cursor-plugin.js:101:20)
    at createDecorations (/Users/krunohrvatinic/ProjectRootDirName/node_modules/y-prosemirror/src/plugins/cursor-plugin.js:85:25)
    at init (/Users/krunohrvatinic/ProjectRootDirName/node_modules/y-prosemirror/src/plugins/cursor-plugin.js:165:16)
    at reconfigure (/Users/krunohrvatinic/ProjectRootDirName/node_modules/prosemirror-state/dist/index.js:868:81)
...
```

The exception happens because `ystate.binding.mapping` is an empty map before **ySyncPlugin** finishes initializing, and `relativePositionToAbsolutePosition` expects a full and initialized map.

### Workaround

I've fixed the issue for myself by patching `cursor-plugin.js`
```diff
export const createDecorations = (state, awareness, awarenessFilter, createCursor, createSelection) => {
   const ystate = ySyncPluginKey.getState(state)
   const y = ystate.doc
   const decorations = []
   if (
     ystate.snapshot != null || ystate.prevSnapshot != null ||
-    ystate.binding === null
+    ystate.binding.mapping.size === 0
   ) {
     // do not render cursors while snapshot is active
     return DecorationSet.create(state.doc, [])
   }
```

This seems like the right thing to do because the null check will never return true. The null check was originally added in 1.1.3, when **ySyncPlugin** initialized binding to `null`. This was changed very recently in 1.2.7 - **ySyncPlugin** now initializes `binding` to `new ProsemirrorBinding`, so binding is never `null`.

### Environment Information
- MacOS, happens in Safari, Firefox and Chrome
 - yjs@13.6.14
 - y-prosemirror@1.2.8
 
I'm using y-prosemirror through Remirror, but am unaffected by #155 because my app waits until Y.Doc is loaded before loading ySyncPlugin
I'm using Liveblocks as the provider.

* [ ] I'm a sponsor 💖
* [ ] This issue is a blocker for my project.

**Comment by dmonad** [comment]
Thanks! I added a fix. This will be part of the next release.

**Comment by khrvati** [comment]
@dmonad Thank you!

If you're touching this part of the code anyway, there is a similar null-check here
https://github.com/yjs/y-prosemirror/blob/f41a3e161448fe23e83a5ad69297274e4de7fb09/src/plugins/cursor-plugin.js#L207
If I got the root cause right, the check is a no-op

Leaving it in won't cause problems because absolutePositionToRelativePosition has a built-in way to handle an empty map
https://github.com/yjs/y-prosemirror/blob/f41a3e161448fe23e83a5ad69297274e4de7fb09/src/lib.js#L82

...but you may want to remove or change it anyway?

**Comment by nyacg** [comment]
As above, we're also using y-prosemirror through tiptap and are on 1.2.9 and still experiencing this error

**Comment by nyacg** [comment]
Any update on this?

**Comment by dmonad** [comment]
The fix is released in y-prosemirror@1.2.10. Please make sure to use the latest version and fill out the issue template when you open another ticket.
